lib/active_support/cache/memcached_store.rb

```diff
@@ -279,7 +292,7 @@ def expiration(options)
 
       def handle_exceptions(return_value_on_error:, on_miss: return_value_on_error)
         yield
-      rescue Memcached::NotFound, Memcached::ConnectionDataExists
+      rescue Memcached::NotFound, Memcached::NotStored, Memcached::ConnectionDataExists
```

Comment:
@dylanahsmith do you have suggestions on how to handle the fact that this is required for the `handle_exceptions`  to handle `on_miss: false`?

IMO `@cache.write("foo", unless_exist: true)` should return false, not raise `Memcached::NotStored`, but I don't know what the ramifications would be if that change were made.

If it's not safe to change the behaviour of `write()`, then I suspect that we either need to provide `handle_exceptions` the operation name (ugh) so it can rescue `Memcached::NotStored` and re-raise unless the operation is write, *or*, `append` has to remove the use of `handle_exceptions`.

Reply:
Side note: this was the original implementation of the test for cache miss:

```ruby
  def test_append_with_cache_miss
    refute(@cache.append('foo', 'bar'))
  end
```

...and all tests passed because `nil` satisfies `refute`.

Reply:
handle_exceptions is just doing a rescue, so we could just do that explicitly in `append` using

```ruby
        handle_exceptions(return_value_on_error: nil) do
          begin
            instrument(:append, name) do
              @connection.append(normalized_key, value)
            end
            true
          rescue Memcached::NotStored
            false
          end
        end
```

so we don't have to modify `handle_exceptions`